detox/src/devices/common/drivers/android/tools/HashHelper.js

```diff
@@ -0,0 +1,33 @@
+const SUPPORTED_HASH_TYPES = {
+  MD5: 'MD5'
+}
+
+const HASH_PATH = '/data/local/tmp/detox';
+
+class HashHelper {
+  constructor(adb, hashXfer) {
+    this._adb = adb;
+    this._hashXfer  = hashXfer;
+  }
+
+  generateHash(path, hashType = SUPPORTED_HASH_TYPES.MD5) {
+    const crypto = require('crypto');
+    switch (hashType) {
+      case SUPPORTED_HASH_TYPES.MD5:
+      default:
```

Comment:
I'm not a fan of this approach, because `generateHash(path, 'fubar')` would in fact result in a valid MD5 hash as well (i.e. no indication there was an invalid usage of the helper).